runtime/cudaq/algorithms/get_kernel_name.h

```diff
@@ -0,0 +1,69 @@
+/****************************************************************-*- C++ -*-****
```

Comment:
this is broader than algorithms, so I would suggest not putting it here (maybe in utils or qis). I also would suggest something more generic like `kernel_utils.h`.